src/Symfony/Component/HttpKernel/Kernel.php

```diff
@@ -344,6 +344,30 @@ public function setAnnotatedClassCache(array $annotatedClasses)
     }
 
     /**
+     * Return a list of classes to be cached away when the container is build.
+     *
+     * If a class contains annotations, add it to getAnnotatedClassesToCompile()instead.
+     *
+     * @return string[]
+     */
+    public function getClassesToCompile()
```

Comment:
This is mostly for new apps (which should use 7.0) so imo it doesn't make sense to allow that if that's not worth it in 7.0 but of course as @nicolas-grekas said, at worst we'll only have to deprecate it later.